src/components/map/DesktopMap.jsx

```diff
@@ -240,26 +239,29 @@ class DesktopMap extends PureComponent {
                 plugin={PluginLeaflet}
                 onReady={() => { if (loading) setLoading(false); }}
               >
-                {layers.map((l, i) => (
-                  <Layer
-                    {...l}
-                    key={l.id}
-                    opacity={l.opacity}
-                    zIndex={1000 - i}
-                    {...l.params && { params: l.params }}
-                    {...l.sqlParams && { sqlParams: l.sqlParams }}
-                    {...l.decodeParams && { decodeParams: l.decodeParams }}
-                    {...l.interactionConfig && {
-                      interactivity: ['carto', 'cartodb'].includes(l.provider)
-                        ? (l.interactionConfig.output || []).map(o => o.column)
-                        : true
-                    }}
-                    events={{
-                      ...mapMode === 'analysis' && { click: (e) => { this.handlePoint(e, _map); } },
-                      ...mapMode === 'view' && { click: (e) => { this.handleClickMap(e); } }
-                    }}
-                  />
-                  ))}
+                {layers.map((layer, i) => {
+                  const l = { ...layer };
```

Comment:
This creates a copy of the object, right? Could you explain why this is needed to a react newbie like me?

Reply:
@lbraun You aren't missing anything, this can go back to how it was.  I was destructuring stuff out of it to log, and probably just ended up leaving this in place, thinking it was how it was originally. I don't think there's a good reason to clone the object here.

Reply:
Reverted this to how it was